asset/asset_structure.go

```diff
@@ -86,7 +111,16 @@ func traverseLocationalTree(node LocationalNode, projectId string, locationalPar
 	return createdCnt, nil
 }
 
-func traverseFunctionalTree(node FunctionalNode, projectId string, locationalParentAssetId, functionalParentAssetId *int32) (createdCnt int, err error) {
+func traverseFunctionalTree(
+	node FunctionalNode,
+	projectId string,
+	locationalParentAssetId,
+	functionalParentAssetId *int32,
+	upsertingData bool,
+	ts *time.Time,
+	clientReference *string,
```

Comment:
This is getting crazy, too much parameters :D. I would create a params struct at this point. But it's unexported functions, so we can do spring cleaning later.